src/index.ts

```diff
@@ -422,8 +422,20 @@ export default runExtension({
       handler:
         ({ proccessBlockText, variables }) =>
         (arg, format = "(({uid}))") => {
-          const queryUid = variables[arg] || arg;
-          const parentUid = extractRef(queryUid);
+          const queryRef = variables[arg] || arg;
+          const aliasOrPage = window.roamAlphaAPI.data.fast.q(
+            `[:find ?uid :where [?b :block/uid ?uid] [or-join [?b] 
+                 [and [?b :block/string ?s] [[clojure.string/includes? ?s "{{query block:${queryRef}}}"]] ]
+                 ${getQueryPages(extensionAPI).map(
+                   (p) =>
+                     `[and [?b :node/title ?t] [[re-pattern "^${p.replace(
+                       /\*/,
+                       queryRef
+                     )}$"] ?regex] [[re-find ?regex ?t]]]`
```

Comment:
why not a straight comparison `[and [?b :node/title "^${p.replace(/\*/, queryRef)}$"]]`?

I also think we'll want to add a branch for comparing the title directly: `[and [?b :node/title "${queryRef}"]]`

Reply:
What is the use case for comparing the title directly?